execution_engine/src/core/engine_state/executable_deploy_item.rs

```diff
@@ -320,7 +320,12 @@ impl ExecutableDeployItem {
                 error::Error::Exec(execution::Error::NoSuchMethod(entry_point_name.to_owned()))
             })?;
 
-        if protocol_data.system_contracts().contains(&contract_hash) {
+        let mut system_hashes = tracking_copy
+            .borrow_mut()
+            .get_system_contracts(correlation_id)?
+            .into_values();
+
+        if system_hashes.any(|x| x == contract_hash) {
```

Comment:
Better to use find than any as find would stop on first `true` condition.

```suggestion
        if system_hashes.find(|&value| value == contract_hash) {
```

Reply:
`any()` does, too: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any

> `any()` is short-circuiting; in other words, it will stop processing as soon as it finds a `true`, given that no matter what else happens, the result will also be `true`.

Reply:
Right, it's the same thing, basically. Let's leave it with `any`

Reply:
I applied the suggestion, however, `find() -> Option<&ContractHash>` and tailing it with an `.is_some()` is returning a lint warning, the compiler suggests `any` in favor of find

Reply:
TIL. Thanks, @darthsiroftardis